feature(Implement Link vouchers from Invoices and Cash Payments)

lomamech opened the pull request:
- Update drop down On Invoice and Cash Payment Registry, add voucher
  Registry link
- Update API for find voucher and filter for invoice_uuid and cash_uuid

closes #2146

lomamech committed:
feature(Implement Link vouchers from Invoices and Cash Payments)
- Update drop down On Invoice and Cash Payment Registry, add voucher
  Registry link
- Update API for find voucher and filter for invoice_uuid and cash_uuid

jniles commented on client/src/modules/invoices/registry/templates/action.cell.html:
```diff
@@ -22,11 +22,17 @@
       </a>
     </li>
     <li>
+      <a data-method="view-invoice"  href ui-sref="vouchers({ filters : [{ key : 'invoice_uuid', value : row.entity.uuid, displayValue: row.entity.reference }]})">
```
I think this should be 'view-voucher'.

jniles commented on client/src/modules/cash/payments/templates/action.cell.html:
```diff
@@ -21,10 +21,16 @@
       </a>
     </li>
+    <li ng-show="!row.entity.is_caution">
+      <a data-method="view-invoice"  href ui-sref="vouchers({ filters : [{ key : 'cash_uuid', value : row.entity.uuid, displayValue: row.entity.reference }]})">
```
This `data-method` should be 'view-voucher'.

jniles reviewed (commented):
@lomamech, just a couple clean up items.

Could you implement an end to end test for these?  I've added some tools to make it much easier (and safer) to write end to end tests.  You can see an example [here](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/test/end-to-end/vouchers/vouchers.spec.js#L21).

Basically, you should pick an invoice with a credit note, and write something like this:
```js
 // this is an example invoice with a credit note
const row = new GridRow('IV.TPA.4');

// this line opens the drop down menu on that grid row.
row.dropdown().click(); 

// this line you will have to implement.
row.goToVoucher().click();
```

The `goToVoucher()` method should be a method on [GridRow](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/test/end-to-end/shared/GridRow.js#L37) and look like this:
```js
goToVoucher() {
  return this.menu.$('[data-method="view-voucher"]');
}
```

The same thing can go for payments and that link.

lomamech committed:
test(link voucher from Invoice and Cash Registry)
- Update Data for test set current date for invoice and cash Data
- fix error set good data-method view-voucher
- Reset data for test E2E on invoice registry search
- Reset date for test E2E on cash registry search
- Add function goToVoucher in shared/gridRow
- Implete the E2E test check link voucher from Invoice and CashRegistry

jniles commented on test/end-to-end/shared/GridRow.js:
```diff
@@ -49,6 +49,11 @@ class GridRow {
   }
+
+  goToVoucher() {
+    return this.menu.$('[data-method="view-voucher"]');
```
:+1:

jniles reviewed (approved):
LGTM.

bors r+
